packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js

```diff
@@ -20,34 +22,64 @@ export default async (req, res) => {
     .retrieveByCartId(cartId)
     .catch(() => undefined)
 
+  let responseCode
+
   // handle payment intent events
   switch (event.type) {
     case "payment_intent.succeeded":
       if (order && order.payment_status !== "captured") {
-        await orderService.capturePayment(order.id)
+        await manager.transaction(async (manager) => {
+          await orderService.withTransaction(manager).capturePayment(order.id)
+        })
       }
       break
-    //case "payment_intent.canceled":
-    //  if (order) {
-    //    await orderService.update(order._id, {
-    //      status: "canceled",
-    //    })
-    //  }
-    //  break
-    case "payment_intent.payment_failed":
-      // TODO: Not implemented yet
-      break
     case "payment_intent.amount_capturable_updated":
       if (!order) {
-        await cartService.setPaymentSession(cartId, "stripe")
-        await cartService.authorizePayment(cartId)
-        await orderService.createFromCart(cartId)
+        // Find and use the same idempotency that has been created and used for the cart that is used for that payment
+        let idempotencyKey
+        try {
+          idempotencyKey = await manager.transaction(
+            async (transactionManager) => {
+              const idempotencyKeyRepository =
+                transactionManager.getCustomRepository(IdempotencyKeyRepository)
+              return await idempotencyKeyRepository
```

Comment:
@olivermrbl here i used the repo, but we might. Need to move it somewhere else if we go with that approach

Reply:
I think we need to rethink this a bit. Let's huddle and get back with a comment here 👍